dbms/src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -922,7 +949,8 @@ BlockInputStreams MergeTreeDataSelectExecutor::spreadMarkRangesAmongStreamsWithO
     {
         size_t need_marks = min_marks_per_stream;
 
-        BlockInputStreams streams_per_thread;
+        std::vector<OutputPort *> streams_per_thread;
+        Processors pipe;
```

Comment:
What is pipe? As I can see from code below we pipe can have multiple inputs and multiple outputs, and `Transforms` inside one pipe seems independent?

Reply:
I think that I need to separate a separate interface for Pipe.